lib/net/ldap.rb

```diff
@@ -540,6 +540,9 @@ def initialize(args = {})
     @base = args[:base] || DefaultTreebase
     @force_no_page = args[:force_no_page] || DefaultForceNoPage
     @encryption = args[:encryption] # may be nil
+    if !@encryption.nil? and !@encryption.is_a?(Hash)
```

Comment:
Can we switch this to `@encryption && !@encryption.is_a?(Hash)`? This makes the nil check implicit, and the `&&` operator is higher precedence than `and` (doesn't matter in this case, but I prefer it when used in a conditional)